### ISSUE TITLE
Flatpak exports point towards a binary in the nix store, breaking on flatpak updates.

### ISSUE THREAD

**Issue by GabTheBab** [NixOS/nixpkgs#156664]
### Describe the bug
Flatpak exports point to the nix store path of flatpak, which will break on updates. EDIT: This seems to only happen when going through gnome-software, might happen with other flatpak frontends. Using the CLI does give correct paths in exports.

### Steps To Reproduce
1. Install a flatpak package via gnome-software
2. Look inside its desktop file, its pointing to a binary in the store

### Expected behavior
There was a fix in nixpkgs to have it directly use "flatpak run", which doesn't seem to work when using gnome software.

### Notify maintainers
@jtojnar 
### Metadata

```console
➜ nix-shell -p nix-info --run "nix-info -m"
 - system: `"x86_64-linux"`
 - host os: `Linux 5.15.12-xanmod1-tt, NixOS, 22.05 (Quokka)`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.6.0pre20211217_6e6e998`
 - channels(root): `"nixos"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos
```

**Comment by jtojnar** [comment]
Relevant change https://github.com/NixOS/nixpkgs/pull/141913

**Comment by vlinkz** [comment]
Should be fixed by #222678 on the latest nixos-unstable releleases
